src/query/service/src/sql/planner/semantic/type_check.rs

```diff
@@ -929,7 +930,12 @@ impl<'a> TypeChecker<'a> {
         let mut arg_types = vec![];
 
         for argument in arguments {
-            let box (arg, arg_type) = self.resolve(argument, None).await?;
+            let box (arg, mut arg_type) = self.resolve(argument, None).await?;
```

Comment:
Do we need to handle this in `self.resolve()`?

Reply:
No